include/votca/xtp/lanczossolver.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2009-2019 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+#ifndef __VOTCA_TOOLS_LANCZOS_SOLVER_H
+#define __VOTCA_TOOLS_LANCZOS_SOLVER_H
+
+#include <chrono>
+#include <iostream>
+#include <stdexcept>
+
+#include <boost/format.hpp>
+#include <votca/xtp/eigen.h>
+#include <votca/xtp/shiftinvert_operator.h>
+#include <votca/xtp/logger.h>
+#include <votca/xtp/matrixfreeoperator.h>
+
+#include <Spectra/GenEigsRealShiftSolver.h>
+#include <Spectra/GenEigsSolver.h>
```

Comment:
If you include spectra headers in an public header, you will need to install spectra as well.